### PR TITLE
Refacto store:  userSlice

### LINKED ISSUE #5173: Refacto store: Slice user
**Description and goal**

- [ ] add useSlice
- [ ]  add selectors

user    
|— isLogged 
|— toLogin
|— loginError 
|— serverError 
|— username
|— accessToken
|— account

### PR DESCRIPTION
**Pull request by Uriel-Sautron**
close #5173 
- Creating userSlice
- Create userSelector and use it in the application
- Creating tests for userSlice

### CODE REVIEW

**Comment by kmer2016** [comment]
Well done ! Just, In the "common" folder, I would have called the reducer store folder because it actually contains a bit more than just reducers (reducer, slice, selectors)

File: front/src/main/home.jsx [hunk 1]
```diff
@@ -8,16 +8,14 @@ import stdcmImg from 'assets/pictures/home/stdcm.svg';
...
-import { useSelector } from 'react-redux';
 import { useTranslation } from 'react-i18next';
 
 export default function Home() {
-  const user = useSelector((state) => state.user);
   const { t } = useTranslation('home/home');
 
   return (
     <>
-      <NavBarSNCF appName="OSRD" username={user.username} logo={logo} />
```
Comment (anisometropie) [inline front/src/main/home.jsx:20]: this wasn’t used anymore ?

File: front/src/reducers/__tests__/userReducer.spec.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+describe('userReducer', () => {
+  test('should return the initial state', () => {
+    const store = createStore(userInitialState);
+    const userState = store.getState().user;
+    expect(userState).toEqual(userInitialState);
+  });
+
+  test('should handle loginSuccess', () => {
+    const store = createStore(userInitialState);
+    const action = loginSuccess({ accessToken: '', username: 'Test userSlice' });
```
Comment (anisometropie) [inline front/src/reducers/__tests__/userReducer.spec.ts]:
```suggestion
    const action = loginSuccess({ accessToken: 'fake_token', username: 'Test userSlice' });
```
Here use something that’s not going to be a default value, or something that can end up here by chance

File: front/src/reducers/__tests__/userReducer.spec.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    store.dispatch(action);
+    const userState = store.getState().user;
+    expect(userState).toEqual({
+      isLogged: true,
+      toLogin: false,
+      loginError: false,
+      serverError: false,
+      username: 'Test userSlice',
+      accessToken: '',
```
Comment (anisometropie) [inline front/src/reducers/__tests__/userReducer.spec.ts]:
```suggestion
      accessToken: 'fake_token',
```

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/reducers/__tests__/userReducer.spec.ts (hunk 2 above)
Comment (Uriel-Sautron) [reply front/src/reducers/__tests__/userReducer.spec.ts]: Thank you, done !!

File: front/src/reducers/__tests__/userReducer.spec.ts (hunk 3 above)
Comment (Uriel-Sautron) [reply front/src/reducers/__tests__/userReducer.spec.ts]: Done

**Review by anisometropie — APPROVED** [review]

**Review by Math-R — APPROVED** [review]
LGTM

**Pull request merged**